.pre-commit-config.yaml

```diff
@@ -1,41 +1,216 @@
 exclude: demisto_sdk/tests/test_files/.*|.circleci/config.yml|demisto_sdk/commands/.*/tests/test_files/.*
 repos:
-    - repo: https://github.com/charliermarsh/ruff-pre-commit
-      rev: "v0.0.262"
-      hooks:
-        - id: ruff
-          args: [--fix, --exit-non-zero-on-fix]
-    - repo: https://github.com/psf/black
-      rev: "22.12.0"
-      hooks:
-          - id: black
-    - repo: https://github.com/pre-commit/pre-commit-hooks
-      rev: v4.4.0
-      hooks:
-          - id: trailing-whitespace
-          - id: end-of-file-fixer
-          - id: check-docstring-first
-            exclude: demisto_sdk/commands/init/templates/.*
-          - id: check-json
-            exclude: .vscode/.*
-          - id: check-yaml
-            language_version: python3
-          - id: name-tests-test
-            exclude: test_tools.py|demisto_sdk/commands/download/tests/tests_env/.*|demisto_sdk/commands/create_artifacts/tests/data|demisto_sdk/commands/common/content/tests/objects/pack_objects/script/script_test/TestNotUnifiedScript|demisto_sdk/commands/common/content/tests/objects/pack_objects/integration/integration_test/TestNotUnifiedIntegration|demisto_sdk/commands/test_content/tests/DemistoClientMock
-    - repo: https://github.com/demisto/demisto-sdk
-      rev: v1.16.0
-      hooks:
-          - id: mypy
+- repo: https://github.com/charliermarsh/ruff-pre-commit
+  rev: "v0.0.262"
+  hooks:
+  - id: ruff
+    args: [--fix, --exit-non-zero-on-fix]
+- repo: https://github.com/python-poetry/poetry
+  rev: "1.5.1"
+  hooks:
+  - id: poetry-check
+  - id: poetry-lock
+    args: [--check]
+  - id: poetry-export
+    args: [--without-hashes, "-o", "requirements.txt"]
```

Comment:
Please add `requirements.txt` to `.gitignore` so nobody commits it by mistake

Reply:
it is already in gitignore